src/js/components/locations-configuration/LocationAddress.jsx

```diff
@@ -99,14 +100,12 @@ class LocationAddress extends Component {
   }
 
   saveAddressOfLocation(values, callback) {
-    const valuesAsAddressObject = {
-      address: {
-        ...values,
-      },
-    };
     this.props.showSpinner();
-    const locationUrl = `/api/locations/${this.state.locationId}`;
-    apiClient.post(locationUrl, flattenRequest(valuesAsAddressObject))
+    const payload = Object.keys(values).reduce((acc, curr) => {
+      if (values[curr]) return { address: { ...acc.address, [curr]: values[curr] } };
+      return { address: { ...acc.address, [curr]: '' } };
+    }, { address: {} });
+    apiClient.post(LOCATION_POST(this.state.locationId), payload)
```

Comment:
This doesn't feel right if we don't want to send address at all - right now you always send address with empty strings.

If I am not right, at least add the brackets after `if`, and format the JSON not to have whole JSON in one line please.
```javascript
if (values[curr]) {
  ...
}
```

Reply:
also consider adding a comment of what is going on